moonstone/plot/plot_template.py

```diff
@@ -0,0 +1,134 @@
+from abc import ABC
+import math
+import numpy as np
+import pandas as pd
+import plotly.graph_objects as go
+import plotly.io
+from typing import Union
+
+
+class BaseGraph(ABC):
+
+    def __init__(self, dataframe: pd.DataFrame, plotting_options: dict = None,
+                 show: bool = True, output_file: Union[bool, str] = False):
+        """
+        :param dataframe: pandas dataframe generated with moonstone
+        :param show: set to False if you don't want to show the plot
+        :param output_file: name of the output file
+        :param plotting_options: options of plotting that will override the default setup \n
+                                 [!] Make sure the value given to an argument is of the right type \n
+                                 options allowed : 'log': `bool` ; 'colorbar': `[str, List[str]]` ;
+                                 'tickangle': `[int, float]`
+        """
+        self.df = dataframe
+        self.plotting_options = plotting_options
+
+        if type(show) == bool:
+            self.show = show
+        else:
+            raise ValueError('Error : show value must be a bool, %s given' % type(show).__name__)
+
+        self.output_file = output_file
+        if self.output_file is True:
+            # if no name given for the output file, a generic name is generated
+            if dataframe.name is not None:
+                self.output_file = dataframe.name+"_"+self.__class__.__name__+".html"
+            else:
+                self.output_file = self.__class__.__name__+".html"
+
+    def plot_one_graph(self, title: str, xlabel: str, ylabel: str):
+        """
+        method that plots the graph
+        """
+        # instantiate fig
+
+        # treat every plotting options by updating layout
+
+        # if self.show is True:
+        #    fig.show()
+
+        # if self.output_file:
+        #    plotly.io.write_html(fig, self.output_file)
+        pass
+
+
+class BarGraph(BaseGraph):
```

Comment:
All the methods should be tested with unit tests and automatically called when calling `plot_one_graph()`. Opened issue: https://github.com/motleystate/moonstone/issues/13